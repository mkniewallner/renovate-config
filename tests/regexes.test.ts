import { extractPackageFile } from "renovate/dist/modules/manager/custom/regex";
import { regEx } from "renovate/dist/util/regex";
import { loadFixture, loadRenovateConfiguration } from "./utils";

function regexMatches(target: string, patterns: string[]): boolean {
  return patterns.some((pattern: string) => {
    return regEx(pattern).test(target);
  });
}

describe("Update PEP 440 Python dependencies", () => {
  const regexManager = loadRenovateConfiguration()["customManagers"][0];

  test.each([".pre-commit-config.yaml", "pyproject.toml"])(
    "find dependencies in `%s`",
    (fileName) => {
      expect(extractPackageFile(loadFixture(fileName), fileName, regexManager)).toMatchSnapshot();
    },
  );

  describe("matches regexes patterns", () => {
    test.each([
      [".pre-commit-config.yaml", true],
      ["apre-commit-configayaml", false],
      [".pre-commit-config.yamlfoobar", false],
      ["foo/.pre-commit-config.yaml", false],
      ["pyproject.toml", true],
      ["pyprojectatoml", false],
      ["pyproject.tomlfoobar", false],
      ["foo/pyproject.toml", false],
    ])('regexMatches("%s") === %s', (path, expected) => {
      expect(regexMatches(path, regexManager.fileMatch)).toBe(expected);
    });
  });
});
