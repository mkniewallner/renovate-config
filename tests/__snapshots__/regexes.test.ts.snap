// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Update PEP 440 Python dependencies > find dependencies in \`.pre-commit-config.yaml\` 1`] = `
{
  "datasourceTemplate": "pypi",
  "deps": [
    {
      "currentValue": "==21.9.1",
      "datasource": "pypi",
      "depName": "flake8-bugbear",
      "replaceString": ""flake8-bugbear==21.9.1"  # renovate: pep440-python-dependency
",
    },
    {
      "currentValue": "==0.14.1",
      "datasource": "pypi",
      "depName": "flake8-simplify",
      "replaceString": ""flake8-simplify==0.14.1" # renovate: pep440-python-dependency
",
    },
  ],
  "matchStrings": [
    ""(?<depName>[\\w-]+)(?<currentValue>.+?)",? +# renovate: pep440-python-dependency\\s",
  ],
}
`;

exports[`Update PEP 440 Python dependencies > find dependencies in \`pyproject.toml\` 1`] = `
{
  "datasourceTemplate": "pypi",
  "deps": [
    {
      "currentValue": "==21.3.0",
      "datasource": "pypi",
      "depName": "argon2-cffi",
      "replaceString": ""argon2-cffi==21.3.0",  # renovate: pep440-python-dependency
",
    },
    {
      "currentValue": " == 1.2.1",
      "datasource": "pypi",
      "depName": "arrow",
      "replaceString": ""arrow == 1.2.1", # renovate: pep440-python-dependency
",
    },
    {
      "currentValue": " >= 0.16.0",
      "datasource": "pypi",
      "depName": "uvloop",
      "replaceString": ""uvloop >= 0.16.0" # renovate: pep440-python-dependency
",
    },
  ],
  "matchStrings": [
    ""(?<depName>[\\w-]+)(?<currentValue>.+?)",? +# renovate: pep440-python-dependency\\s",
  ],
}
`;
